faros_event.sh

```diff
@@ -0,0 +1,470 @@
+                            "source": $s
+                        }
+                    }
+                },
+                "commit": {
+                "repository": {
+                    "organization": {
+                        "uid": $vcs_org_uid,
+                        "source": $vcs_source
+                    },
+                    "name": $repo
+                },
+                "sha": $commit_sha
+                }
+            }
+        }'
+    )
+}
+
+function makePipeline() {
+    cicd_Pipeline=$( jq -n \
+        --arg s "$SOURCE" \
+        --arg pipeline_uid "$pipeline_uid" \
+        --arg ci_org_uid "$ci_org_uid" \
+        '{
+            "cicd_Pipeline": {
+                "uid": $pipeline_uid,
+                "organization": {
+                    "uid": $ci_org_uid,
+                    "source": $s
+                }
+            }
+        }'
+    )
+}
+
+function makeApplication() {
+    compute_Application=$( jq -n \
+        --arg app_name "$application_name" \
+        --arg app_platform "$APPLICATION_PLATFORM" \
+        '{
+            "compute_Application": {
+                "name": $app_name,
+                "platform": $app_platform
+            }
+        }'
+    )
+}
+
+function makeBuildEvent() {
+    makeBuild
+    makeBuildCommitAssociation
+    makePipeline
+    makeApplication
+    request_body=$( jq -n \
+        --arg origin "$ORIGIN" \
+        --argjson build "$cicd_Build" \
+        --argjson buildCommit "$cicd_BuildCommitAssociation" \
+        --argjson pipeline "$cicd_Pipeline" \
+        --argjson application "$compute_Application" \
+        '{ 
+            "origin": $origin,
+            "entries": [
+                $build,
+                $buildCommit,
+                $pipeline,
+                $application
+            ]
+        }')
+}
+
+function makeDeploymentEvent() {
+    makeDeployment
+    makeApplication
+    request_body=$( jq -n \
+        --arg origin "$ORIGIN" \
+        --argjson deployment "$cicd_Deployment" \
+        '{ 
+            "origin": $origin,
+            "entries": [
+                $deployment
+            ]
+        }')
+}
+
+function makeFullEvent() {
+    makeDeployment
+    makeBuild
+    makeBuildCommitAssociation
+    makePipeline
+    makeApplication
+    request_body=$( jq -n \
+        --arg origin "$ORIGIN" \
+        --argjson deployment "$cicd_Deployment" \
+        --argjson build "$cicd_Build" \
+        --argjson buildCommit "$cicd_BuildCommitAssociation" \
+        --argjson pipeline "$cicd_Pipeline" \
+        --argjson application "$compute_Application" \
+        '{ 
+            "origin": $origin,
+            "entries": [
+                $deployment,
+                $build,
+                $buildCommit,
+                $pipeline,
+                $application
+            ]
+        }')
+}
+
+function emitToFaros() {
+    echo "Emitting event to Faros..."
+
+    curl -s $faros_api_url/graphs/$faros_graph/revisions \
```

Comment:
are you sure about `-s`? we might want to print the status code and the response in case of the failure

Reply:
I made the http response handling better. Let me know if it is what you had in mind.